xmodule/capa_block.py

```diff
@@ -2145,7 +2187,12 @@ def rescore(self, only_if_higher=False):
         event_info['orig_total'] = orig_score.raw_possible
         try:
             self.update_correctness()
-            calculated_score = self.calculate_score()
+            grading_strategy_handler = GradingStrategyHandler(
+                self.grading_strategy,
+                self.score_history,
+                self.max_score(),
+            )
+            calculated_score = grading_strategy_handler.get_score()
```

Comment:
Should we move this piece of code to its own method? so we can re-use it elsewhere